pages/_app.tsx

```diff
@@ -1,8 +1,33 @@
-import '../styles/globals.css'
-import type { AppProps } from 'next/app'
+import "../styles/globals.css";
+import type { AppProps } from "next/app";
+import { useState } from "react";
 
 function MyApp({ Component, pageProps }: AppProps) {
-  return <Component {...pageProps} />
+  /**
+   * This hook measures the time between push of start and stop button on the pages
+   * "Homepage" and "Test". Thus it is called in the app component.
+   */
+
+  const [startingTime, setStartingTime] = useState(0);
+  const [stoppingTime, setStoppingTime] = useState(0);
+
+  // Time specification in milliseconds, has to be converted into minutes. However, it is still a decimal value
+  // which again has to be converted into a proper hexadecimal seconds count when displayed in the result tab.
+  const calculatedTime = (stoppingTime - startingTime) / 60000;
+  const minutes = Math.floor(calculatedTime);
+  const seconds = Math.round(
+    (calculatedTime - Math.floor(calculatedTime)) * 60
+  );
+
+  return (
+    <Component
+      {...pageProps}
+      minutes={minutes}
+      seconds={seconds}
```

Comment:
I've though about this for some time now.

I think you calculation is the simplest solution for now. If you need to do anything more complex you could use a library like `date-fns`. But for this small scope you'll be fine.

That said, i think i'd move the calculation of `minutes` and `seconds` down into the result component and only pass down `calculatedTime`.
This way you avoid passing down multiple derived values. Also your score calculations will be easier in the future because you can just to math with calculatedTime inside the result component.

Another thing: If you wanted to be a bit more fancy. You could consider making this a custom hook using react context. You could make a very nice API using that pattern.

Kinda like this:
```tsx
const Test = () => {
  const {time, start, stop} = useStopwatch();

  return <Button onClick={start}>Start timer!</Button>
}
```